modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -576,6 +579,60 @@ def raise_exception_if_value_present(val)
           detail: "The #{val} is required for direct deposit."
         )
       end
+
+      private
+
+      # checking begin > end
+      def begin_date_is_not_after_end_date?(begin_date, end_date)
+        # see what format each date is in
+        begin_date_has_day = date_has_day?(begin_date)
+        end_date_has_day = date_has_day?(end_date)
+        # determine how to compare, being = is ok
+        if (begin_date_has_day && end_date_has_day) || (!begin_date_has_day && !end_date_has_day) # same format
+          begin_date > end_date
+        else # mixed formats on dates
+          begin_date_not_after_end_date_with_mixed_format_dates?(begin_date, end_date)
+        end
+      end
+
+      def begin_date_not_after_end_date_with_mixed_format_dates?(begin_date, end_date)
+        # figure out which one has the day and remove it
+        if date_has_day?(begin_date)
+          begin_date = remove_chars(begin_date.dup)
+        elsif date_has_day?(end_date)
+          end_date = remove_chars(end_date.dup)
+        end
+        Date.strptime(begin_date, '%m-%Y') > Date.strptime(end_date, '%m-%Y') # = is ok
+      end
+
+      def date_is_valid_against_now_after_check_on_format?(date)
```

Comment:
Not necessary for my approval, but consider rethinking some of these method names. It's a little difficult to determine what they're meant to do (from the outside looking in, at least)